### PR TITLE
Add "billing_information" RBAC resource 

### PR DESCRIPTION
**Pull request by alex-kovoy**
This PR adds billing information RBAC resource and exposes several Auth Service methods to make it possible to register custom GRPC services (for auth plugin).

It's counterpart https://github.com/gravitational/teleport.e/pull/219

### CODE REVIEW

**Review by Joerger — APPROVED** [review]

File: api/types/constants.go [hunk 1]
```diff
@@ -190,6 +190,9 @@ const (
 	// KindMFADevice is an MFA device for a user.
 	KindMFADevice = "mfa_device"
 
+	// KindBillingInformation represents access to cloud billing features
+	KindBillingInformation = "billing_information"
```
Comment (r0mant) [inline api/types/constants.go:194]: Nit: Maybe it's just me but I prefer short names unless they compromise meaning, so I'd just name this "billing" (information suffix seems redundant).

**Review by r0mant — APPROVED** [review]

File: lib/web/sessions.go [hunk 2]
```diff
@@ -79,6 +79,14 @@ func (c *SessionContext) String() string {
 	)
 }
 
+// GetAuthClient returns auth client
+func (c *SessionContext) GetAuthClient() (*auth.Client, error) {
```
Comment (russjones) [inline lib/web/sessions.go:83]: What's the difference between this and `GetClient` in the same file?

Comment (alex-kovoy) [reply lib/web/sessions.go]: @russjones removed

**Review by russjones — APPROVED** [review]
Bot.

**Review by klizhentas — APPROVED** [review]

**Comment by klizhentas** [comment]
@alex-kovoy any reason why you returned *auth.Client instead of the interface here?

File: lib/plugin/registry.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+// NewRegistry creates an instance of the Registry
+func NewRegistry() Registry {
+	return &registry{
+		plugins: make(map[string]Plugin),
+	}
+}
+
+type registry struct {
+	plugins map[string]Plugin
```
Comment (alex-kovoy) [inline lib/plugin/registry.go:53]: This map is not going to be used concurrently as plugin registrations happen during the initialization time. 

Btw, @klizhentas i am not happy with the package name `plugin` as it overlaps with other plugins (data). What if I name it as `enterprise` implying enterprise only extensions?

Comment (klizhentas) [reply lib/plugin/registry.go:53]: I think either `plugin` or `extensions` is fine. I would stay away from `enterprise`

**Review by klizhentas — APPROVED** [review]

**Pull request merged**